udata/core/discussions/api.py

```diff
@@ -0,0 +1,130 @@
+    'title': fields.String(description='The discussion title', required=True),
+    'user': fields.Nested(user_ref_fields, description='The discussion author', required=True),
+    'created': fields.ISODateTime(description='The discussion creation date', readonly=True),
+    'closed': fields.ISODateTime(description='The discussion closing date', readonly=True),
+    'closed_by': fields.String(attribute='closed_by.id', description='The user who closed the discussion', readonly=True),
+    'discussion': fields.Nested(message_fields),
+    'url': fields.UrlFor('api.discussion', description='The discussion API URI', readonly=True),
+})
+
+comment_discussion_fields = api.model('DiscussionResponse', {
+    'comment': fields.String(description='The comment to submit', required=True),
+    'close': fields.Boolean(description='Is this a closing response. Only subject owner can close')
+})
+
+discussion_page_fields = api.model('DiscussionPage', fields.pager(discussion_fields))
+
+parser = api.parser()
+parser.add_argument('closed', type=bool, default=False, location='args', help='Filter closed discussions')
+parser.add_argument('for', type=str, location='args', action='append', help='Filter issues for a given subject')
+parser.add_argument('page', type=int, default=1, location='args', help='The page to fetch')
+parser.add_argument('page_size', type=int, default=20, location='args', help='The page size to fetch')
+
+
+@ns.route('/<id>/', endpoint='discussion')
+@api.doc(model=discussion_fields)
+class DiscussionAPI(API):
+    '''
+    Single Discussion Model API (Read and update).
+    '''
+    @api.doc('get_discussion')
+    @api.marshal_with(discussion_fields)
+    def get(self, id):
+        '''Get an discussion given its ID'''
+        discussion = Discussion.objects.get_or_404(id=id)
+        return discussion
+
+    @api.secure
+    @api.doc('comment_discussion')
+    @api.expect(comment_discussion_fields)
+    @api.response(403, 'Not allowed to close this discussion')
+    @api.marshal_with(discussion_fields)
+    def post(self, id):
+        '''Add comment and optionnaly close an discussion given its ID'''
+        discussion = Discussion.objects.get_or_404(id=id)
+        form = api.validate(DiscussionCommentForm)
+        message = Message(
+            content=form.comment.data,
+            posted_by=current_user.id
+        )
+        discussion.discussion.append(message)
+        close = form.close.data
+        if close:
+            CloseDiscussionPermission(discussion).test()
+            discussion.closed_by = current_user._get_current_object()
+            discussion.closed = datetime.now()
+        discussion.save()
+        if close:
+            on_discussion_closed.send(discussion, message=message)
+        else:
+            on_new_discussion_comment.send(discussion, message=message)
+        return discussion
+
+
+@ns.route('/', endpoint='all_discussions')
```

Comment:
Without another  listing endpoint, `api.discussions` may be better than `api.all_discussions`